elrond.md

```diff
@@ -918,6 +951,40 @@ TODO: Implement [reserved keys and read-only runtimes](https://github.com/Elrond
          </locals>
 ```
 
+### Crypto API
+
+```k
+    // extern int32_t sha256(void* context, int32_t dataOffset, int32_t length, int32_t resultOffset);
+    rule <instrs> hostCall("env", "sha256", [ i32 i32 i32 .ValTypes ] -> [ i32 .ValTypes ])
+               => #memLoad(DATAOFFSET, LENGTH)
+               ~> #sha256FromBytesStack
+               ~> #memStoreFromBytesStack(RESULTOFFSET)
+               ~> #dropBytes
+               ~> i32.const 0
+                  ...
+         </instrs>
+         <locals>
+           0 |-> <i32> DATAOFFSET
+           1 |-> <i32> LENGTH
+           2 |-> <i32> RESULTOFFSET
+         </locals>
+
+    // extern int32_t keccak256(void *context, int32_t dataOffset, int32_t length, int32_t resultOffset);
+    rule <instrs> hostCall("env", "keccak256", [ i32 i32 i32 .ValTypes ] -> [ i32 .ValTypes ])
+               => #memLoad(DATAOFFSET, LENGTH)
+               ~> #keccakFromBytesStack
+               ~> #memStoreFromBytesStack(RESULTOFFSET)
+               ~> #dropBytes
+               ~> i32.const 0
+                  ...
+         </instrs>
+         <locals>
+           0 |-> <i32> DATAOFFSET
+           1 |-> <i32> LENGTH
+           2 |-> <i32> RESULTOFFSET
+         </locals>
```

Comment:
Looks like you could easily simplify these two rules into one:

```
syntax InternalInstrs ::= #hashMemory ( HashBytesStackInstr )
```
where the `HashBytesStackInstr` is either `#sha256FromBytesStack` or `#keccakFromBytesStack`